File: 13305/13305.py
import sys
sys = sys.stdin.readline
N = int(input())
dist = list(map(int,input().strip().split()))
price = list(map(int,input().strip().split()))
minPrice, ans = price[0], price[0] * dist[0]
for i in range(1,N-1):
    minPrice = min(minPrice, price[i])
    ans += dist[i] * minPrice
print(ans)

File: BOJ/13305/13305.cpp
#include <cstdlib>
#include <iostream>
using namespace std;

int main()
{
    int N, dist[100005] = {0,}, price[100005] = {0,};
    scanf("%d", &N);
    for (int i = 0; i < N - 1; i++) scanf("%d", &dist[i]);
    for (int i = 0; i < N; i++) scanf("%d", &price[i]);
    
    int minPrice = price[0];
    long long ans = (long long)price[0] * dist[0];
    for (int i = 1; i < N - 1; i++)
    {
        minPrice = price[i] < minPrice ? price[i] : minPrice;
        ans += (long long)dist[i] * minPrice;
    }
    printf("%lld \n", ans);
    return 0;
}